### ISSUE TITLE
Threaded Instancing: Control is not drawn if Control.visibility is changed from false to true during _ready()

### ISSUE THREAD

**Issue by jamie-pate** [godotengine/godot#42347]
**Godot version:**

v3.1-stable..3.2.3-stable (probably < 3.1 also, I couldn't open the reproduction in 3.0.6)

**OS/device including version:**

windows, mac, linux
Pop!_OS 20.04 LTS (like ubuntu 20.4)
OpenGL ES 3.0 Renderer: GeForce GTX 1070 with Max-Q Design/PCIe/SSE2
nvidia-driver-440

**Issue description:**

When calling PackedScene.instance() in a thread, then adding it to the scene and an ancestor `Control.visible` property is changed from `false` to `true` during `_ready()` it will not be drawn.
![thread-instanced-scene-glitch-2020-09-25_16 00 56-5fps](https://user-images.githubusercontent.com/682658/94322835-7b533400-ff48-11ea-93f4-6a01c7faf626.gif)

**Steps to reproduce:**
Run the reproduction.

Relevant code:

```Python
        var SomePackedScene: PackedScene = load('res://SomePackedScene.tsn')
	var instance = yield(BgWorker.new().instance_scene(SomePackedScene), 'completed')
	$"2dPanel".add_child(instance)
```

`BgWorker.gd`:

```Python
extends Reference

signal _thread_done(status)

var _thread: Thread

func instance_scene(scene: PackedScene) -> Node:
	assert(!_thread)
	_thread = Thread.new()
	_thread.start(self, '_thread_load', scene)
	var result: Node = yield(self, '_thread_done')
	_thread.wait_to_finish()
	_thread = null
	return result

func _thread_load(scene: PackedScene) -> void:
	var status: int = OK
	var instance := scene.instance()
	call_deferred('_thread_done', instance)

func _thread_done(result: Node) -> void:
	emit_signal('_thread_done', result)
```

**Minimal reproduction project:**

[gui_in_3d_repro2.zip](https://github.com/godotengine/godot/files/5285731/gui_in_3d_repro2.zip)

**Comment by jamie-pate** [comment]
Does `call_deferred` here effectively place the signal on the main thread? If not that could be the problem

**Comment by jamie-pate** [comment]
A workaround is to yield for an idle frame and re-set the visibility then.

```Python
	for n in nodes:
		n.visible = false
	yield(get_tree(), 'idle_frame')
	yield(get_tree(), 'idle_frame')
	for n in nodes:
		n.visible = n.name == name
```

**Comment by jamie-pate** [comment]
Seems to be fixed in godot 4.3-beta2
